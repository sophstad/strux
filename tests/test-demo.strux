int main() {

    :( Tree Demo ):
    BSTree::int tree = new BSTree::int(2,3,1);
    print("Original Tree");
    tree.show();
    tree.add(-1);
    print("Adding -1...");
    tree.show();
    tree.add(-5);
    print("Adding -5...");
    tree.show();
    tree.add(-2);
    print("Adding -2...");
    tree.show();
    tree.add(50);
    print("Adding 50...");
    tree.show();
    tree.add(30);
    print("Adding 30...");
    tree.show();
    tree.add(6);
    print("Adding 6...");
    tree.show();
    tree.add(4);
    print("Adding 4...");
    tree.show();
    tree.delete(4);
    print("Deleting 4...");
    tree.show();
    tree.delete(6);
    print("Deleting 6...");
    tree.show();
    tree.delete(-5);
    print("Deleting -5...");
    tree.show();
    tree.delete(2);
    print("Deleting 2...");
    tree.show();
    tree.delete(1);
    print("Deleting 1...");
    tree.show();
    tree.delete(50);
    print("Deleting 50...");
    tree.show();
    tree.delete(3);
    print("Deleting 3...");
    tree.show();
    tree.delete(-1);
    print("Deleting -1...");
    tree.show();
    tree.delete(-2);
    print("Deleting -2...");
    tree.show();
    tree.delete(30);
    print("Deleting 30...");
    tree.show();

    num[5] n = [1.1, 4.2, 5.2, -2.32, 32.2];
    BSTree::num numTree = new BSTree::num();
    print("");
    print("");
    print("New num tree...");
    numTree.show();
    print("");
    
    int i = 0;
    for (i = 0; i < 5; i++) {
        print("Adding...");
        print(n[i]);
        print("");
        numTree.add(n[i]);
        numTree.show();
    }
    for(i = 4; i >= 0; i--) {
        print("Deleting...");
        print(n[i]);
        print("");
        numTree.delete(n[i]);
        numTree.show();
    }


    :( Quicksort Demo ):
    print("");
    print("");
    num[10] numbers = [1.11, 42.21, 32.322, -34.3, 34.2, 96.1, -3.4, 232.1, 2.2, 99.1];
    print("Before sorting numbers...");
    numbers.show();
    numbers.quickSort();
    print("After sorting numbers...");
    numbers.show();
    print("");
    print("Showing quicksort on strings...");
    string[6] a = ["kofi", "stephen", "millie", "sophie", "josh", "chang"];
    a.showQuickSort();
    
    return 0;
}