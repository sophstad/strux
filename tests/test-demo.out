Original Tree
      (0002)                                                                    
   *----*------*                                                                
(0001)      (0003)                                                              

--------------------------------------------------------------------------------
Adding -1...
            (0002)                                                              
         *----*------*                                                          
      (0001)      (0003)                                                        
   *-----*                                                                      
(-001)                                                                          

--------------------------------------------------------------------------------
Adding -5...
                  (0002)                                                        
               *----*------*                                                    
            (0001)      (0003)                                                  
         *-----*                                                                
      (-001)                                                                    
   *-----*                                                                      
(-005)                                                                          

--------------------------------------------------------------------------------
Adding -2...
                        (0002)                                                  
                     *----*------*                                              
                  (0001)      (0003)                                            
               *-----*                                                          
            (-001)                                                              
   *-----------*                                                                
(-005)                                                                          
  *------*                                                                      
      (-002)                                                                    

--------------------------------------------------------------------------------
Adding 50...
                        (0002)                                                  
                     *----*------*                                              
                  (0001)      (0003)                                            
               *-----*          *------*                                        
            (-001)                  (0050)                                      
   *-----------*                                                                
(-005)                                                                          
  *------*                                                                      
      (-002)                                                                    

--------------------------------------------------------------------------------
Adding 30...
                        (0002)                                                  
                     *----*------*                                              
                  (0001)      (0003)                                            
               *-----*          *------------*                                  
            (-001)                        (0050)                                
   *-----------*                       *-----*                                  
(-005)                              (0030)                                      
  *------*                                                                      
      (-002)                                                                    

--------------------------------------------------------------------------------
Adding 6...
                        (0002)                                                  
                     *----*------*                                              
                  (0001)      (0003)                                            
               *-----*          *------------------*                            
            (-001)                              (0050)                          
   *-----------*                             *-----*                            
(-005)                                    (0030)                                
  *------*                             *-----*                                  
      (-002)                        (0006)                                      

--------------------------------------------------------------------------------
Adding 4...
                        (0002)                                                  
                     *----*------*                                              
                  (0001)      (0003)                                            
               *-----*          *------------------------*                      
            (-001)                                    (0050)                    
   *-----------*                                   *-----*                      
(-005)                                          (0030)                          
  *------*                                   *-----*                            
      (-002)                              (0006)                                
                                       *-----*                                  
                                    (0004)                                      

--------------------------------------------------------------------------------
Deleting 4...
                        (0002)                                                  
                     *----*------*                                              
                  (0001)      (0003)                                            
               *-----*          *------------------*                            
            (-001)                              (0050)                          
   *-----------*                             *-----*                            
(-005)                                    (0030)                                
  *------*                             *-----*                                  
      (-002)                        (0006)                                      

--------------------------------------------------------------------------------
Deleting 6...
                        (0002)                                                  
                     *----*------*                                              
                  (0001)      (0003)                                            
               *-----*          *------------*                                  
            (-001)                        (0050)                                
   *-----------*                       *-----*                                  
(-005)                              (0030)                                      
  *------*                                                                      
      (-002)                                                                    

--------------------------------------------------------------------------------
Deleting -5...
                  (0002)                                                        
               *----*------*                                                    
            (0001)      (0003)                                                  
         *-----*          *------------*                                        
      (-001)                        (0050)                                      
   *-----*                       *-----*                                        
(-002)                        (0030)                                            

--------------------------------------------------------------------------------
Deleting 2...
                  (0003)                                                        
               *----*------------*                                              
            (0001)            (0050)                                            
         *-----*           *-----*                                              
      (-001)            (0030)                                                  
   *-----*                                                                      
(-002)                                                                          

--------------------------------------------------------------------------------
Deleting 1...
            (0003)                                                              
         *----*------------*                                                    
      (-001)            (0050)                                                  
   *-----*           *-----*                                                    
(-002)            (0030)                                                        

--------------------------------------------------------------------------------
Deleting 50...
            (0003)                                                              
         *----*------*                                                          
      (-001)      (0030)                                                        
   *-----*                                                                      
(-002)                                                                          

--------------------------------------------------------------------------------
Deleting 3...
            (0030)                                                              
         *-----*                                                                
      (-001)                                                                    
   *-----*                                                                      
(-002)                                                                          

--------------------------------------------------------------------------------
Deleting -1...
      (0030)                                                                    
   *-----*                                                                      
(-002)                                                                          

--------------------------------------------------------------------------------
Deleting -2...
(0030)                                                                          

--------------------------------------------------------------------------------
Deleting 30...
Tree is empty!


New num tree...
Tree is empty!

Adding...
1.100000

(1.10)                                                                          

--------------------------------------------------------------------------------
Adding...
4.200000

(1.10)                                                                          
  *------*                                                                      
      (4.20)                                                                    

--------------------------------------------------------------------------------
Adding...
5.200000

(1.10)                                                                          
  *------*                                                                      
      (4.20)                                                                    
        *------*                                                                
            (5.20)                                                              

--------------------------------------------------------------------------------
Adding...
-2.320000

      (1.10)                                                                    
   *----*------*                                                                
(-2.32      (4.20)                                                              
              *------*                                                          
                  (5.20)                                                        

--------------------------------------------------------------------------------
Adding...
32.200000

      (1.10)                                                                    
   *----*------*                                                                
(-2.32      (4.20)                                                              
              *------*                                                          
                  (5.20)                                                        
                    *------*                                                    
                        (32.20                                                  

--------------------------------------------------------------------------------
Deleting...
32.200000

      (1.10)                                                                    
   *----*------*                                                                
(-2.32      (4.20)                                                              
              *------*                                                          
                  (5.20)                                                        

--------------------------------------------------------------------------------
Deleting...
-2.320000

(1.10)                                                                          
  *------*                                                                      
      (4.20)                                                                    
        *------*                                                                
            (5.20)                                                              

--------------------------------------------------------------------------------
Deleting...
5.200000

(1.10)                                                                          
  *------*                                                                      
      (4.20)                                                                    

--------------------------------------------------------------------------------
Deleting...
4.200000

(1.10)                                                                          

--------------------------------------------------------------------------------
Deleting...
1.100000

Tree is empty!


Before sorting numbers...
[1.110000, 42.210000, 32.322000, -34.300000, 34.200000, 96.100000, -3.400000, 232.100000, 2.200000, 99.100000]
After sorting numbers...
[-34.300000, -3.400000, 1.110000, 2.200000, 32.322000, 34.200000, 42.210000, 96.100000, 99.100000, 232.100000]

Showing quicksort on strings...
=====================================================================================
At this step:
current array: [ kofi stephen millie sophie josh chang ]
pivot swapped: kofi,chang
array after swap: [ chang stephen millie sophie josh kofi ]
pivot swapped: millie,kofi
array after swap: [ chang stephen kofi sophie josh millie ]
strings swapped: josh,kofi
array after swap: [ chang stephen josh sophie kofi millie ]
pivot is kofi
strings swapped: stephen,josh
array after swap: [ chang josh stephen sophie kofi millie ]
strings swapped: stephen,kofi
array after swap: [ chang josh kofi sophie stephen millie ]
strings swapped: millie,sophie
array after swap: [ chang josh kofi millie stephen sophie ]
=====================================================================================
At this step:
current array: [ chang josh kofi millie stephen sophie ]
pivot is josh
=====================================================================================
At this step:
current array: [ chang josh kofi millie stephen sophie ]
pivot is chang
=====================================================================================
At this step:
current array: [ chang josh kofi millie stephen sophie ]
pivot swapped: stephen,sophie
array after swap: [ chang josh kofi millie sophie stephen ]
pivot is sophie
=====================================================================================
QuickSort complete! Final Result: [ chang josh kofi millie sophie stephen ]
